src/runtime/internal/Component.ts

```diff
@@ -212,16 +212,19 @@ if (typeof HTMLElement === 'function') {
 	};
 }
 
-export class SvelteComponent {
+export class SvelteComponent<
+	Props extends Record<string, any> = any,
+	Events extends Record<string, any> = any
+> {
 	$$: T$$;
-	$$set?: ($$props: any) => void;
+	$$set?: ($$props: Partial<Props> & Record<string, any>) => void;
```

Comment:
I am curious, why is it `Partial<Props> & Record<string, any>`, not `Partial<Props>` ?

Reply:
The existence of `$$restProps` signaled to me that passing props other than those defined is possible and not an error, so I made this typing more loose. Same for the constructor typing.

Reply:
Imho by default the constraints should be strict but a component should be able to declare that it accepts any properties

Reply:
That's a good point, people can always type `& Record<string, any>` themselves, I made it more strict.